### PR TITLE
Enable ftsb_redisearch related benchmarks on CI benchmarks

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR:
- Adds 6 more test cases to the CI benchmarks triggering ( [sample green perf job](https://app.circleci.com/pipelines/github/RediSearch/RediSearch/4438/workflows/18bc052a-6bc4-4bb6-afd0-3299d0082ca8/jobs/27298) )
- Enable the `PROFILE=1` env var build adjustment so that we retain the frame pointer and debug info when building the artifacts for profiling.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,6 +111,10 @@ ifneq ($(SAN),)
 CMAKE_SAN += -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++
 endif
 
+ifeq ($(PROFILE),1)
+CMAKE_SAN += -DPROFILE=ON
```
Comment (rafie) [inline Makefile:115]: Is this related to Clang sanitizer?

File: cmake/redisearch_cflags.cmake [hunk 2]
```diff
@@ -15,6 +15,10 @@ SET(RS_COMMON_FLAGS "${RS_COMMON_FLAGS} -fPIC -Werror=implicit-function-declarat
 SET(RS_COMMON_FLAGS "${RS_COMMON_FLAGS} -pthread")
 SET(RS_COMMON_FLAGS "${RS_COMMON_FLAGS} -fno-strict-aliasing")
 
+IF (PROFILE)
+    SET(RS_COMMON_FLAGS "${RS_COMMON_FLAGS} -g -fno-omit-frame-pointer")
```
Comment (rafie) [inline cmake/redisearch_cflags.cmake:19]: Can we trigger a `PROFILE` flag on cmake (i.e. `-DPROFILE=1` or something) and make the resolution in `cmake.debug`?

File: tests/ci.benchmarks/.gitignore [hunk 3]
```diff
@@ -4,3 +4,4 @@
 datasets/*.rdb
 *.rdb
 binaries
+profile_*
```
Comment (rafie) [inline tests/ci.benchmarks/.gitignore:7]: In which directory those file actually created in?

File: tests/ci.benchmarks/defaults.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      - commit
+    timemetric: "$.StartTime"
+    metrics:
+      - "$.Tests.Overall.rps"
+      - "$.Tests.Overall.avg_latency_ms"
+      - "$.Tests.Overall.p50_latency_ms"
+      - "$.Tests.Overall.p95_latency_ms"
+      - "$.Tests.Overall.p99_latency_ms"
+      - "$.Tests.Overall.max_latency_ms"
+      - "$.Tests.Overall.min_latency_ms"
```
Comment (rafie) [inline tests/ci.benchmarks/defaults.yml:15]: \n

File: tests/ci.benchmarks/requirements.txt [hunk 5]
```diff
@@ -1 +1 @@
-redisbench_admin>=0.3.1
+redisbench_admin>=0.4.0
```
Comment (rafie) [inline tests/ci.benchmarks/requirements.txt:1]: Would we like those requirements to be installed as part of `system-setup.py`?

File: tests/ci.benchmarks/.gitignore (hunk 3 above)
Comment (filipecosta90) [reply tests/ci.benchmarks/.gitignore:7]: > In which directory those file actually created in?
```
tests/ci.benchmarks/profile_*
```
Will adjust it :)

File: tests/ci.benchmarks/requirements.txt (hunk 5 above)
Comment (filipecosta90) [reply tests/ci.benchmarks/requirements.txt:1]: given it has an extended deps list I didn't want to break automation in any way because of this. It's documented on the Readme how to install it.

File: tests/ci.benchmarks/.gitignore (hunk 3 above)
Comment (filipecosta90) [reply tests/ci.benchmarks/.gitignore:7]: already saw that it's in this folder gitignore. I believe this is the correct way to do it (meaning no change). Agree?

Comment (rafie) [reply tests/ci.benchmarks/.gitignore:7]: yep.

**Review by rafie — APPROVED** [review]
